core/arch/arm/mm/sp_mem.c

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2021, Arm Limited. All rights reserved.
+ */
+#include <assert.h>
+#include <bitstring.h>
+#include <ffa.h>
+#include <kernel/spinlock.h>
+#include <mm/mobj.h>
+#include <mm/sp_mem.h>
+
+#define NUM_SHARES	64
+
+static bitstr_t bit_decl(share_bits, NUM_SHARES);
+static unsigned int sp_mem_lock = SPINLOCK_UNLOCK;
+
+/* mem_shares stores all active FF-A shares. */
+SLIST_HEAD(sp_mem_head, sp_mem);
+static struct sp_mem_head mem_shares = SLIST_HEAD_INITIALIZER(sp_mem_head);
+const struct mobj_ops mobj_sp_ops;
```

Comment:
static?

Reply:
We can't have mobj_sp_ops static cause it is defined as weak.

Reply:
deserves a inline comment IMO `/* Weak instance of mobj_sp_ops mandates it is not static */`, or like.

Reply:
Done